pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -210,6 +210,15 @@ func (m *kubeGenericRuntimeManager) generateContainerResources(pod *v1.Pod, cont
 	}
 }
 
+// generateUpdatePodSandboxResourcesRequest generates platform specific (linux) podsandox resources config for runtime
+func (m *kubeGenericRuntimeManager) generateUpdatePodSandboxResourcesRequest(sandboxID string, pod *v1.Pod) *runtimeapi.UpdatePodSandboxResourcesRequest {
```

Comment:
@tallclair This function cannot be inline because the functions **convertOverheadToLinuxResources** and **calculateSandboxResources** are platform specific, and break on typechecker.